### PR TITLE
Cleanup ADIOS2_ROOT for some E3SM machines

### PR DESCRIPTION
**Pull request by dqwu**
So far, we will only test the new ADIOS IO type on the following
machines: Perlmutter, Frontier, Summit, and ANL GCE workstations.

Additionally, the ADIOS2 library has been upgraded from version
2.8.3 to 2.9.1 on these machines.

Existing ADIOS2_ROOT configurations have been removed on specific
machines, including Chrysalis, Anvil, Bebop, and Compy.

[BFB]

### CODE REVIEW

**Comment by dqwu** [comment]
@ndkeen ADIOS support has been added for the default intel compiler on pm-cpu. Also, aocc/3.2.0 module is no longer available on pm-cpu (replaced by aocc/4.0.0).

**Review by ndkeen — APPROVED** [review]
It looks like all that is happening on NERSC machine is an update to version of adios. So approved, but have not tested.

**Comment by dqwu** [comment]
@jayeshkrishna 
For ADIOS 2.9.x, we have the following build errors with the default scorpio release 1.4.1
```
E3SM/externals/scorpio/tools/adios2pio-nm/adios2pio-nm-lib.cxx:2147:41: error: no member named 'DebugON' in namespace 'adios2'
    adios2::ADIOS adios(w_comm, adios2::DebugON);
                                ~~~~~~~~^
```
This known issue has been fixed in scorpio release 1.4.2 and 1.5.0

Maybe you can create a PR similar to #5983 for E3SM master branch? We can merge that PR first, then this PR.

**Comment by dqwu** [comment]
@jgfouca Please hold this PR before we have another PR to upgrade scorpio from 1.4.1 to 1.4.2 (or 1.5.0) for E3SM aster.branch.

**Comment by jgfouca** [comment]
@dqwu , OK, just let me know when it's ready.

File: cime_config/machines/config_machines.xml
Comment (jayeshkrishna) [inline cime_config/machines/config_machines.xml:230]: Was this change required for the new version of ADIOS (2.9.1)?

Comment (dqwu) [reply cime_config/machines/config_machines.xml:230]: It was required for amdclang compiler to build on pm-cpu (3.2.0 module is no longer available).

**Comment by dqwu** [comment]
> @dqwu , OK, just let me know when it's ready.

@jgfouca The default scorpio submodule has been upgraded to 1.4.2 for E3SM next and master branch.

**Comment by dqwu** [comment]
@jgfouca The PR has been rebased to resolve merge conflicts with latest next or master branch.

**Comment by dqwu** [comment]
@jgfouca Please make sure that the latest (rebased) feature branch is checked out before merging it to E3SM next branch.

**Comment by jgfouca** [comment]
OK, merged to next.

**Pull request merged**
